passes/pmgen/peepopt.cc

```diff
@@ -60,11 +58,8 @@ struct PeepoptPass : public Pass {
 
 		if (!genmode.empty())
 		{
-			initbits.clear();
-			rminitbits.clear();
-
 			if (genmode == "shiftmul")
-				GENERATE_PATTERN(peepopt_pm, shiftmul);
+				GENERATE_PATTERN(peepopt_pm, shiftmul_right);
```

Comment:
```suggestion
				log_error("Abort in %s:%d: peepopt-shiftmul does not implement a generate option\n", __FILE__, __LINE__)
```

Doesn't this require a generate block in the pattern to actually work?

Running it ends with:
```
ERROR: pmgen generator is stuck: 10000 iterations with no matching module generated.
```

I think the most relevant PR is #1298.

Reply:
I don't know, I never looked into the generate stuff. Then maybe we should remove the generate flag altogether, I don't think muldiv has any generate blocks either.